nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,756 @@
+          "CAP_NET_BIND_SERVICE"
+          "CAP_NET_RAW"
+          "CAP_SETGID"
+          "CAP_SETUID"
+          "CAP_SYS_CHROOT"
+          "CAP_CHOWN"
+          "CAP_FOWNER"
+          "CAP_DAC_OVERRIDE"
+          "CAP_AUDIT_WRITE"
+          "CAP_SYS_ADMIN"
+        ];
+        AmbientCapabilities = [
+          "CAP_NET_RAW"
+          "CAP_NET_BIND_SERVICE"
+        ];
+      };
+    };
+
+    systemd.services.homebridge-config-ui-x = {
+      description = "Homebridge Config UI X";
+      wants = [ "network-online.target" ];
+      after = [
+        "syslog.target"
+        "network-online.target"
+        "homebridge.service"
+      ];
+      requires = [ "homebridge.service" ];
+      wantedBy = [ "multi-user.target" ];
+      path = with pkgs; [
+        # Tools listed in homebridge's installation documentations:
+        # https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Arch-Linux
+        nodejs_22
+        nettools
+        gcc
+        gnumake
+        # Required for access to systemctl and journalctl
+        systemd
+        # Required for access to sudo
+        "/run/wrappers"
+        # Some plugins need bash to download tools
+        bash
+      ];
+
+      environment = {
+        HOMEBRIDGE_CONFIG_UI_TERMINAL = "1";
+        DISABLE_OPENCOLLECTIVE = "true";
+        # Required or homebridge will search the global npm namespace
+        UIX_STRICT_PLUGIN_RESOLUTION = "1";
+      };
+
+      # Settings found from standalone mode docs and hb-service code
+      # https://github.com/homebridge/homebridge-config-ui-x/wiki/Standalone-Mode
+      # https://github.com/homebridge/homebridge-config-ui-x/blob/a12ad881fe6df62d817ecb56f9fc6b7e82b6d078/src/bin/platforms/linux.ts#L714
+      serviceConfig = {
+        Type = "simple";
+        User = "homebridge";
+        PermissionsStartOnly = true;
+        WorkingDirectory = cfg.userStoragePath;
+        EnvironmentFile = mkIf (cfg.environmentFile != null) [ cfg.environmentFile ];
+        ExecStart = "${pkgs.homebridge-config-ui-x}/bin/homebridge-config-ui-x ${args}";
+        Restart = "always";
+        RestartSec = 3;
+        KillMode = "process";
+        CapabilityBoundingSet = [
+          "CAP_IPC_LOCK"
+          "CAP_NET_ADMIN"
+          "CAP_NET_BIND_SERVICE"
+          "CAP_NET_RAW"
+          "CAP_SETGID"
+          "CAP_SETUID"
+          "CAP_SYS_CHROOT"
+          "CAP_CHOWN"
+          "CAP_FOWNER"
+          "CAP_DAC_OVERRIDE"
+          "CAP_AUDIT_WRITE"
+          "CAP_SYS_ADMIN"
+        ];
+        AmbientCapabilities = [
+          "CAP_NET_RAW"
+          "CAP_NET_BIND_SERVICE"
+        ];
+      };
+    };
+
+    # Create a user whose home folder is the user storage path
+    users.users.homebridge = {
+      home = cfg.userStoragePath;
+      createHome = true;
+      group = "homebridge";
+      # Necessary so that this user can run journalctl
+      extraGroups = [ "systemd-journal" ];
+      isSystemUser = true;
+    };
```

Comment:
Add `user`/`group` options to configure this (and use that for the `username` setting in the Homebridge config JSON, if that's what it's for).